packages/theme-check-common/src/liquid-doc/utils.ts

```diff
@@ -107,3 +107,19 @@ export function getValidParamTypes(objectEntries: ObjectEntry[]): Map<string, st
 
   return paramTypes;
 }
+
+export function parseParamType(
+  validParamTypes: Set<string>,
+  value: string,
+): [string, boolean] | undefined {
```

Comment:
Are you open to returning a structured type here?

Reply:
We could also name the tuple to make it more obvious that the second boolean means `isArray`

```
): [pseudoType: PseudoType, isArray: boolean] | undefined
```